src/compiler/crystal/macros/interpreter.cr

```diff
@@ -76,13 +76,13 @@ module Crystal
 
     def initialize(@program : Program,
                    @scope : Type, @path_lookup : Type, @location : Location?,
-                   @vars = {} of String => ASTNode, @block : Block? = nil, @def : Def? = nil,
+                   @vars = Hash(String | Int32, ASTNode).new, @block : Block? = nil, @def : Def? = nil,
                    @in_macro = false)
       @str = IO::Memory.new(512) # Can't be String::Builder because of `{{debug}}`
       @last = Nop.new
     end
 
-    def define_var(name, value)
+    def define_var(name : String | Int32, value : ASTNode) : Nil
```

Comment:
Where does that `Int32` come from? You're adding calls to this method with `*_arg.name` which should surely be `String`.